cmd/cli/setup/service_manager.go

```diff
@@ -0,0 +1,274 @@
+package setup
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// CommandExecutor interface for running system commands
+type CommandExecutor interface {
+	Output(name string, args ...string) ([]byte, error)
+	Run(name string, args ...string) error
+}
+
+// RealCommandExecutor executes real system commands
+type RealCommandExecutor struct{}
```

Comment:
is this used elsewhere? if not, just make it a private struct called commandExecutor . I don't love "Real" as syntax for describing an implementation